src/librustc/sir.rs

```diff
@@ -102,7 +104,74 @@ impl SirFuncCx {
 
     /// Converts a MIR statement to SIR, appending the result to `bb`.
     pub fn codegen_statement(&mut self, bb: ykpack::BasicBlockIndex, stmt: &mir::Statement<'_>) {
-        // FIXME: Nothing is implemented yet.
-        self.push_stmt(bb, ykpack::Statement::Unimplemented(format!("{:?}", stmt)));
+        match stmt.kind {
+            mir::StatementKind::Assign(box (ref place, ref rvalue)) => {
+                let assign = self.lower_assign_stmt(place, rvalue);
+                self.push_stmt(bb, assign);
+            },
+            _ => self.push_stmt(bb, ykpack::Statement::Unimplemented(format!("{:?}", stmt)))
+        }
+    }
+
+    fn lower_assign_stmt(&self, lvalue: &mir::Place<'_>, rvalue: &mir::Rvalue<'_>) -> ykpack::Statement {
+        let lhs = ykpack::Local(lvalue.local.as_u32());
+        let rhs = self.lower_rvalue(rvalue);
+        ykpack::Statement::Assign(ykpack::Place::from(lhs), rhs)
+    }
+
+    fn lower_rvalue(&self, rvalue: &mir::Rvalue<'_>) -> ykpack::Rvalue {
+        match rvalue {
+            mir::Rvalue::Use(mir::Operand::Copy(place)) |
+            mir::Rvalue::Use(mir::Operand::Move(place)) => {
+                let local = ykpack::Local(place.local.as_u32());
```

Comment:
Shall we make a `lower_local()` function so that if we ever diverge from having a 1-1 mapping, we do'nt have to look in a million places for where locals are created? It also gives us an opportunity to add a comment about the 1-1 mapping.

Reply:
Fixed in 92d7e50. Will need squashing.